sdk/examples/intkey_cxx/int_key_transaction_processor.cpp

```diff
@@ -0,0 +1,292 @@
+        : std::runtime_error(error)
+    {}
+};
+
+
+// utility function to provide copy conversion from vector of bytes
+// to a stl string container.
+void ToString(std::string* out, const std::vector<std::uint8_t>& in) {
+    out->resize(in.size());
+    const char* data = reinterpret_cast<const char*>(&in[0]);
+    out->assign(data, data+in.size());
+}
+
+// utility function to provide copy conversion from stl string container
+// to a vector of bytes.
+void ToVector(std::vector<std::uint8_t>* out, const std::string& in) {
+    out->resize(in.length());
+    memcpy(&((*out)[0]), in.data(), in.length());
+}
+
+// Handles the processing of a IntKey transactions.
+class IntKeyApplicator:  public sawtooth::TransactionApplicator {
+ public:
+    IntKeyApplicator(sawtooth::TransactionUPtr txn,
+        sawtooth::GlobalStateUPtr state) :
+        TransactionApplicator(std::move(txn), std::move(state)),
+            address_mapper(INTKEY_NAMESPACE) { }
+
+    void Apply() {
+        LOG4CXX_DEBUG(logger, "IntKeyApplicator::Apply");
+
+        const std::string& raw_data = this->txn->payload();
+        std::vector<uint8_t> data_vector(raw_data.begin(), raw_data.end());
+        json intkey_cmd = json::from_cbor(data_vector);
+
+        if (!intkey_cmd.is_object()) {
+            throw InvalidTransaction("CBOR Object as the encoded command");
+        }
+        auto verb_it = intkey_cmd.find("Verb");
+        if (verb_it == intkey_cmd.end()) {
+            throw InvalidTransaction("Verb is required");
+        }
+        std::string verb = *verb_it;
+
+        auto name_it = intkey_cmd.find("Name");
+        if (name_it == intkey_cmd.end()) {
+            throw InvalidTransaction("Name is required");
+        }
+        std::string name = *name_it;
+        if (name.length() == 0 || name.length() >= 20) {
+            throw InvalidTransaction("Name is invalid, name muste be between" \
+                " 1 and 20 characters in length");
+        }
+
+        auto value_it = intkey_cmd.find("Value");
+        if (value_it == intkey_cmd.end()) {
+            throw InvalidTransaction("Value is required");
+        }
+        int value = (*value_it).get<uint32_t>();
+
+        if (verb == "set") {
+            this->DoSet(name, value);
+        } else if (verb == "inc") {
+            this->DoInc(name, value);
+        } else if (verb == "dec") {
+            this->DoDec(name, value);
+        } else {
+            std::stringstream error;
+            error << "invalid Verb: '" << verb << "'";
+            throw InvalidTransaction(error.str());
+        }
+    }
+
+ private:
+    sawtooth::AddressMapper address_mapper;
+
+ private:
+    std::string MakeAddress(const std::string& name) {
+        return this->address_mapper.MakeAddress(name);
+    }
+
+    // Handle an IntKey 'set' verb action. This sets a IntKey value to
+    // the given value and
+    void DoSet(const std::string& name, int value) {
```

Comment:
Short answer is they are not but there will be a follow on task for the c++ sdk along with the other sdks to bring them inline with the intkey spec.